### PR TITLE
chore: workflow dispatch releases

### PR DESCRIPTION
**Pull request by Aslemammad**

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (patak-dev) [inline .github/workflows/ci.yml:17]: Instead of doing this with a workflow dispatch, I think it is better to do a scheme like we have in Vite. We have a release script that bumps the version in main and commits that (together with the changelog but not needed here if you don't want) and tags that commit with `vX.Y.Z`. And then we have an github workflow that publishes each tag that matches that format. See https://github.com/vitejs/vite/blob/main/.github/workflows/publish.yml (we also have another workflow that does a github release on listening to the same event, but that is also not necessary if you don't want). This is a easier flow because you directly release from your CLI with a commit and every published commit is clearly marked in GitHub with a tag.

Comment (Aslemammad) [reply .github/workflows/ci.yml:17]: Yea I was inspired by vite's workflow, and more importantly, the vite-ecosystem-ci workflow! 

To be honest, I think publishing from workflow_dispatches, exactly next to cloudflare deploys would be way better. And then we can create tags (i'll add it in next commits then, I agree).

what do you think?

Comment (patak-dev) [reply .github/workflows/ci.yml:17]: I think it is better to release from a tag instead like we do in Vite because then you force yourself to only publish properly tagged versions. I think it is useful to have this tagged commit also be the one that is doing the bump. So all is connected.

Comment (Aslemammad) [reply .github/workflows/ci.yml:17]: I see, you mean just a `git push origin tag v0.0.18` would trigger a release?

Comment (Aslemammad) [reply .github/workflows/ci.yml:17]: where would the change of the package.json version would be handled?

Comment (patak-dev) [reply .github/workflows/ci.yml:17]: In Vite, we have a `release` script: https://github.com/vitejs/vite/blob/main/scripts/release.ts

So we do `pnpm release` and that will:
- let us choose if we want to release a patch, or a beta for a minor, etc. We have all rules encoded so we can only release what make sense. But you can use something simpler here that just bumps the version.
- it also generate the changelog, but you don't need to if you don't want
- it then commits this changeset + pushes the proper tag

so you only do pnpm release, and that triggers everything, and it ends up in the publish step awaiting for approval, and we will check before approving at that point (instead of in the commit)

Comment (Aslemammad) [reply .github/workflows/ci.yml]: just resolved, let me know what you think!

Comment (Aslemammad) [reply .github/workflows/ci.yml]: just resolved, let me know what you think!

Comment (Aslemammad) [reply .github/workflows/ci.yml]: just resolved, let me know what you think!

Comment (patak-dev) [reply .github/workflows/ci.yml]: I think you still need a `release` npm command that would bump, commit, push, and tag, no? Or do you plan to do all this by hand each time? I'm ok with either, just checking.

Comment (Aslemammad) [reply .github/workflows/ci.yml]: Hmm, I think we can do that but don't we block direct pushes to main? 

So we cannot do things with one command alone. Wdyt?

Comment (patak-dev) [reply .github/workflows/ci.yml]: We can unblock direct pushes to main if that is an issue. We would still have the approval flow to decide if that push will end up or not published, so it is fine.

Comment (Aslemammad) [reply .github/workflows/ci.yml]: just pushed a new commit, let me know what you think.

Comment (patak-dev) [reply .github/workflows/ci.yml]: I'm good with this if vite releases works for you. It is very opinionated for the packages in the Vite repo, so probably better to use something more common, but let's go with this for now.

Comment (Aslemammad) [reply .github/workflows/ci.yml]: yea it works for our case, not heavily tested though since we need to try all the workflow in the main branch.

**Review by AmirSa12 — APPROVED** [review]

**Pull request merged**
